### ISSUE TITLE
Allow toViewPosition() and toSourcePosition() variants that project to the left

### ISSUE THREAD

**Issue by holtgrewe** [seqan/seqan#528]
Currently, we project to the right when hitting a gap with toViewPosition() and toSourcePosition() in Gaps.

We need a second variant that projects to the left.

**Comment by rrahn** [comment]
@holtgrewe If I get this right, then toViewPosition does not need a right/left version, as there is always a unique view position to any source position. The other way around it makes sense, as a view position in a gap is between 2 source anchors can be projected to either the left or the right anchor. 
If there is nothing else intended I would just update the toSourcePosition version.

**Comment by holtgrewe** [comment]
Consider the case below. I believe `view-to-source(5)` is ambiguous as well as `source-to-view(10)`.

```
                   1    5
         0    5    0    5
view     ABCD---HIJKLMOPQ
source   ABCDEFGHI----OPQ
```

**Comment by rrahn** [comment]
Well then I don't understand which functions are meant to be updated?
I only know of toViewPosition and toSourcePosition which works on Gap data structures.
Here a sourcePosition can be mapped to a viewPosition which is always well defined, as there must be a view position for every source position. This is not the case for the other way around, where a gap has no source position. So the example above does not really clarifies the issue here?
it would mean that ```source-to-view(10)``` returns 14 (there would not be any gaps in the source, would it?)
Like this:

```
view:  0123456789
src:   01  23  45
seq:   AC--GT--AC
```

**Comment by holtgrewe** [comment]
```
view:  0123456789
src:   012    345
seq:   ACG----TAC
```

For source to gap consider the two following cases:

1. "Limit gaps object to `source-to-view([0, 3))`"
2. "Give me view coordinates of the character at source pos 3"

**Comment by rrahn** [comment]
Fixed by #1360 

toViewPosition with left projection can be solved as:
```cpp
auto pos = toViewPosition(gaps, srcPos);
auto cGaps = countGaps(gaps, pos, LeftOfViewPos());
auto viewPos = pos - cGaps;
```
If required we can add a convenience function to do just that, but for now this should be fine.
